src/SDPoptimize.jl

```diff
@@ -193,150 +134,46 @@ function solve_DP(model::SPModel,
-                    sampling_size = param.monteCarloSize
-                    samples = [sampling(law,t) for i in 1:sampling_size]
-                    probas = (1/sampling_size)
-                else
-                    sampling_size = law[t].supportSize
-                    samples = law[t].support
-                    probas = law[t].proba
-                end
-
-                for w = 1:sampling_size
-                    w_sample = samples[:, w]
-                    proba = probas[w]
-                    next_state = model.dynamics(t, x, u, w_sample)
-
-
-                    if model.constraints(t, next_state, u, w_sample)
-
-                        count_admissible_w = count_admissible_w + proba
-                        ind_next_state = real_index_from_variable(next_state, x_bounds, x_steps)
-                        next_V = Vitp[ind_next_state...]
-                        current_cost = model.costFunctions(t, x, u, w_sample)
-                        expected_V_u += proba*(current_cost + next_V)
-
-                    end
-                end
-
-                if (count_admissible_w>0)
-
-                    next_V = next_V / count_admissible_w
-
-                    if (expected_V_u < expected_V)
-
-                        expected_V = expected_V_u
-                        optimal_u = u
-
-                    end
-                 end
-            end
-            ind_x = index_from_variable(x, x_bounds, x_steps)
-
-            V[ind_x..., t] = expected_V
+function compute_V_given_t(sampling_size, samples, probas, u_bounds, x_bounds,
+                                x_steps, x_dim, product_states, product_controls,
+                                dynamics, constraints, cost, V, Vitp, t, info_struc)
+
+    if info_struc == "DH"
+        @sync @parallel for indx in 1:length(product_states)
+            SDPutils.compute_V_given_x_t_DH(sampling_size, samples,
+                                            probas, u_bounds, x_bounds,
+                                            x_steps, x_dim, product_controls,
+                                            dynamics, constraints, cost, V, Vitp,
+                                            t, product_states[indx])
+        end
+    elseif info_struc == "HD"
+        @sync @parallel for indx in 1:length(product_states)
+            SDPutils.compute_V_given_x_t_HD(sampling_size, samples, probas,
+                                            u_bounds, x_bounds, x_steps, x_dim,
+                                            product_controls, dynamics,
+                                            constraints, cost, V, Vitp,
+                                            t, product_states[indx])
         end
+    else
+        warn("Information structure should be HD or DH")
     end
-    return V
 end
 
-
 """
-Value iteration algorithm to compute optimal value functions in
-the Hazard Decision (HD) case
+Value iteration algorithm to compute optimal value functions
```

Comment:
Value iteration is often used to refer to an infinite horizon optimization problem.